code/modules/ai/action_states/action_state.dm

```diff
@@ -0,0 +1,69 @@
+//A singleton action state; AI components are added to the list and a subsystem will periodically call the process() for this
+//AI components will also GetTargetDir() from the action_state their connected to so they can have a direction to move in
+/datum/element/action_state
+	var/list/ai_components = list() //All AI components currently attached to this action state
+
+/datum/element/action_state/New()
+	..()
+	if(!QDELETED(src)) //No duplicates allowed
```

Comment:
This won't do what you think it does, also why no duplicates? elements already handle that themselves